dawn/src/dawn/CodeGen/F90Util.h

```diff
@@ -123,15 +128,7 @@ class FortranInterfaceAPI {
       if(std::get<2>(arg) == 0) {
         ss << "value";
       } else {
-        ss << "dimension(";
-        {
-          std::string sep;
-          for(int c = 0; c < std::get<2>(arg); ++c) {
-            ss << sep << ":";
-            sep = ",";
-          }
-        }
-        ss << ")";
```

Comment:
This part is important. We want to ensure that the rank of the array passed matches our expectations. See for instance https://github.com/dawn-ico/ICON-DSL/blob/6478f5a7601395a9a0c391f309d883a7b5f0888c/icon/src/dsl/nh_diffusion.f90#L26-L38

Reply:
But this part doesn't work. We had to replace the `::` with an asterisks to get valid pointers in FORTRAN in our debugging session.

Reply:
documented the situation as requested on slack